gordon/metrics/ffwd.py

```diff
@@ -0,0 +1,200 @@
+The defaults that may be overridden are shown below.
+
+.. code-block:: ini
+
+    [ffwd]
+    # to identify the service creating metrics
+    key = 'gordon-service'
+
+    # the address of the ffwd daemon (see: UDPClient)
+    ip = '127.0.0.9'
+    port = 19000
+
+    # a scaling factor for timing (see: FfwdTimer)
+    time_unit = 1E9
+
+"""
+
+import asyncio
+import json
+import time
+
+import zope.interface
+
+from gordon import interfaces
+
+
+class UDPClientProtocol(asyncio.DatagramProtocol):
+    """Protocol for sending one-off messages via UDP.
+
+    Args:
+        message (bytes): Message for ffwd agent.
+    """
+    def __init__(self, message):
+        self.message = message
+        self.transport = None
+
+    def connection_made(self, transport):
+        """Create connection, use to send message and close.
+
+        Args:
+            transport (asyncio.DatagramTransport): Transport used for sending.
+        """
+        self.transport = transport
+        self.transport.sendto(self.message)
+        self.transport.close()
+
+
+class UDPClient:
+    """Client for sending UDP datagrams.
+
+    Args:
+        ip (str): (optional) Destination IP address (default: 127.0.0.1).
+        port (int): (optional) Destination port (default: 9000).
+        loop (asyncio.AbstractEventLoop impl): (optional) Event loop.
+    """
+
+    DEFAULT_IP = '127.0.0.1'
+    DEFAULT_PORT = 19000
+
+    def __init__(self, ip=None, port=None, loop=None):
+        self.ip = ip or self.DEFAULT_IP
+        self.port = port or self.DEFAULT_PORT
+        self.loop = loop or asyncio.get_event_loop()
+
+    async def send(self, metric):
+        """Transform metric to JSON bytestring and send to server.
+
+        Args:
+            metric (dict): Complete metric to send as JSON.
+        """
+        message = json.dumps(metric).encode('utf-8')
+        await self.loop.create_datagram_endpoint(
+            lambda: UDPClientProtocol(message),
+            remote_addr=(self.ip, self.port))
+
+
+@zope.interface.implementer(interfaces.ITimer)
+class FfwdTimer:
+    """Timer which sends UDP messages to FFWD on completion.
+
+    Args:
+        metric (dict): Dict representation of the metric to send.
+        udp_client (UDPClient): A metric sending client.
+        time_unit (number): (optional) Scale time unit for use with
+            time.time(), for example: 1E9 to send nanoseconds.
+    """
+    def __init__(self, metric, udp_client, time_unit=None):
+        self.metric = metric
+        self.udp_client = udp_client
+        self.time_unit = time_unit or 1
+
+    async def __aenter__(self):
+        """Enter context manager to start timing."""
+        await self.start()
+        return self
+
+    async def __aexit__(self, *args):
+        """Exit context manager to stop timing."""
+        await self.stop()
+
+    async def start(self):
+        """Start timer."""
+        self._start_time = time.time()
```

Comment:
Should this also be changed to `time.perf_counter` like mentioned in [here](https://github.com/spotify/gordon/pull/23#discussion_r193820381) in #23?

Reply:
ah yeah I missed that - good catch @matt0bi .  Yeah I think it should be used here too.

Reply:
It definitely could be.  I don't mind updating.